src/main/java/org/terasology/potions/system/DrinkPotionAuthoritySystem.java

```diff
@@ -161,13 +213,51 @@ public void onPotionWithoutGenomeConsumed(DrinkPotionEvent event, EntityRef ref)
                     break;
             }
 
+            // Before actually consuming the potion and applying its effect, perform a final check and apply some
+            // potential modifications to the effect.
             checkDrink(event.getInstigator(), event.getItem(), p, e, pEffect, effectID);
         }
 
+        // Play the potion drink sound.
         audioManager.playSound(Assets.getSound("engine:drink").get(), 1.0f);
+
+        // Get the EntityRef of the potion, and its durability component (if any).
+        EntityRef potion = event.getItem();
+        DurabilityComponent durability = potion.getComponent(DurabilityComponent.class);
+
+        // If the Durability component exists.
```

Comment:
some people might argue about the usefulness of such comments... I don't mind them, but in this case it is very clear from the code what is happening